### PR TITLE
Drop 2to3 in setup.py

### LINKED ISSUE #399: Eliminate 2to3
The use of 2to3 in setup.py means we can't do an editable install.  It looks like it might be very easy to eliminate the use of 2to3.  @WeatherGod, have you looked at this, and if so, do you see any difficulty with it?

Two options:

1) Make the code compatible with 2 and 3 without using 2to3 in setup.py.  It looks like this requires little more than fixing print statements and using six, or a direct conditional, to handle urllib imports.

2) Skip the compatibility and make a 3-only release, matching matplotlib's.

Thoughts?

### PR DESCRIPTION
**Pull request by ocefpaf**
@efiring I'm not sure of the tests are comprehensive enough to give us 100% confidence on this PR but it was actually quite simple and not a lot of code change to eliminate `2to3`.

Closes #399

### CODE REVIEW

File: lib/mpl_toolkits/basemap/__init__.py [hunk 1]
```diff
@@ -4325,17 +4330,17 @@ def wmsimage(self,server,\
         # ypixels not given, find by scaling xpixels by the map aspect ratio.
         if ypixels is None:
             ypixels = int(self.aspect*xpixels)
-        if verbose: print server
+        if verbose: print(server)
```
[the remark this answers is not in the document]
Comment (ocefpaf) [reply lib/mpl_toolkits/basemap/__init__.py]: I'm actually inclined to add

```python
from __future__ import (absolute_import, division, print_function)
```

at the top of all `.py` files. What do you think?

File: lib/mpl_toolkits/basemap/__init__.py [hunk 2]
```diff
@@ -4347,7 +4352,7 @@ def wmsimage(self,server,\
         return self.imshow(imread(io.BytesIO(img.read()),
                            format=format),origin='upper',alpha=alpha,ax=ax)
         # this works for png, but not jpeg
-        #return self.imshow(imread(urllib2.urlopen(img.url),format=format),origin='upper')
+        #return self.imshow(imread(urlopen(img.url),format=format),origin='upper')
```
[the remark this answers is not in the document]
Comment (ocefpaf) [reply lib/mpl_toolkits/basemap/__init__.py]: Done.

File: lib/mpl_toolkits/basemap/__init__.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (WeatherGod) [reply lib/mpl_toolkits/basemap/__init__.py]: Probably wouldn't hurt. Don't forget about the examples in the docs. They can also be used as a form of unit test (mostly, somewhat, kinda...).

[the remark this answers is not in the document]
Comment (ocefpaf) [reply lib/mpl_toolkits/basemap/__init__.py]: > Probably wouldn't hurt. 

:+1: 


> Don't forget about the examples in the docs. They can also be used as a form of unit test (mostly, somewhat, kinda...).

Working on that in a separate PR b/c most of the issues there are not Python 3 compat code but services that are down and API changes in modern `numpy` and `matplotlib`. See https://github.com/ocefpaf/basemap/tree/update_examples for my progress so far. (PR soon...)

**Comment by ocefpaf** [comment]
@WeatherGod anything left to do here? I sent #401 to update the example, following this one, and I have another PR to update `setup.py` a little bit.

**Pull request merged**